FashionConnect_api/app.js

```diff
@@ -46,75 +46,100 @@ app.get('/__vite_ping', (req, res) => {
 
 // Route to get all users
 app.get('/users', async (req, res) => {
-try {
-    const users = await User.findAll();
-    res.json(users);
-} catch (error) {
-    res.status(500).json({ message: error.message });
-}
+    try {
+        const users = await User.findAll();
+        res.json(users);
+    } catch (error) {
+        res.status(500).json({ message: error.message });
+    }
 });
 
 
 // Route to get all categories
 app.get('/categories', async (req, res) => {
-try {
-    const categories = await Category.findAll();
-    res.json(categories);
-} catch (error) {
-    res.status(500).json({ message: error.message });
-}
+    try {
+        const categories = await Category.findAll();
+        res.json(categories);
+    } catch (error) {
+        res.status(500).json({ message: error.message });
+    }
 });
 
 // Route to get all products, with associated category and reviews
 app.get('/products', async (req, res) => {
-try {
-    const products = await Product.findAll({
-    include: [{ model: Category, as: 'category' }, { model: Review, as: 'reviews' }],
-    order: [['createdAt', 'DESC']]
-    });
-    res.json(products);
-} catch (error) {
-    res.status(500).json({ message: error.message });
-}
+    try {
+        const products = await Product.findAll({
+            include: [{ model: Category, as: 'category' }, { model: Review, as: 'reviews' }],
+            order: [['createdAt', 'DESC']]
+        });
+        res.json(products);
+    } catch (error) {
+        res.status(500).json({ message: error.message });
+    }
 });
 
-// Route to get all collections, with associated user and products
-app.get('/collections', async (req, res) => {
-try {
-    const collections = await Collection.findAll({
-    include: [{ model: User, as: 'user' }, { model: Product, as: 'products' }],
-    order: [['createdAt', 'DESC']]
-    });
-    res.json(collections);
-} catch (error) {
-    res.status(500).json({ message: error.message });
-}
+app.get('/products/:productId', async (req, res) => {
+    const { productId } = req.params;
+
+    try {
+        // Fetch the product details from the database using the productId
+        const product = await Product.findOne({ where: { ProductID: productId } });
+
+        if (!product) {
+            return res.status(404).json({ error: 'Product not found' });
+        }
+        const data = product.dataValues;
+        // Return the product details in the response
+        res.json(data);
+
+    } catch (error) {
+        console.error(error);
```

Comment:
You can remove this when you're done debugging :)

Reply:
will do!